### ISSUE TITLE
get_verify_result not present?

### ISSUE THREAD

**Issue by bortzmeyer** [pyca/pyopenssl#997]
OpenSSL has a routine `SSL_get_verify_result` to check the validation status of a connection for which any result was accepted. I do not find it in PyOpenSSL? 

Context: I want to accept any certificate *but* be informed if they are valid. I can do:

```
context.set_verify(OpenSSL.SSL.VERIFY_PEER | OpenSSL.SSL.VERIFY_FAIL_IF_NO_PEER_CERT | \
                               OpenSSL.SSL.VERIFY_CLIENT_ONCE,
                               lambda conn, cert, errno, depth, preverify_ok: True)
```

But then how to retrieve the validation status?

**Comment by facutuesca** [comment]
You can access the validation status from inside the callback. From the [`SSL_CTX_set_verify()` docs](https://www.openssl.org/docs/manmaster/man3/SSL_CTX_set_verify.html#:~:text=Whenever%20a%20verification%20error%20is%20found%2C%20the%20error%20number%20is%20stored%20in%20x509_ctx%20and%20verify_callback%20is%20called%20with%20preverify_ok%3D0):

> Whenever a verification error is found, **the error number is stored in x509_ctx** and verify_callback is called with preverify_ok=0.

From this, we know that the error code is stored in the context, and the value of `preverify_ok` will say if validation failed or not.

In `PyOpenSSL`, when setting the `verify_callback`, you pass a 5-parameter function. From the [`SSL.Context::set_verify()`](https://www.pyopenssl.org/en/latest/api/ssl.html#OpenSSL.SSL.Context.set_verify) docs:
> callback – The optional Python verification callback to use. This should take five arguments: A Connection object, an X509 object, and three integer variables, which are in turn **potential error number**, error depth and **return code**.

In that doc, **return code** is OpenSSL's `preverify_ok` and **potential error number** is the error code stored in the context.
You can see how `PyOpenSSL` wraps your 5-parameter callback with a helper that gets the error code using `X509_STORE_CTX_get_error`, and passes it to your callback as the 3rd parameter:
https://github.com/pyca/pyopenssl/blob/35d1e87fa3d4714ae1c09955ab63e26fd86dab19/src/OpenSSL/SSL.py#L473-L483

So, in short, you can access the validation status in your callback, and then store it externally if you need to use it outside the callback (as suggested by the [OpenSSL docs](https://www.openssl.org/docs/manmaster/man3/SSL_CTX_set_verify.html#:~:text=The%20calling%20process%20can%20however%20retrieve)):
> The calling process can however retrieve the error code of the last verification error using [SSL_get_verify_result(3)](https://www.openssl.org/docs/manmaster/man3/SSL_get_verify_result.html) or **by maintaining its own error storage managed by verify_callback**.

@mhils I think this one can be closed now.
